robot/Cumulus/resources/RecurringDonationsPageObject.py

```diff
@@ -7,62 +7,80 @@
 from datetime import datetime
 from dateutil.relativedelta import relativedelta
 
+
 @pageobject("Listing", "npe03__Recurring_Donation__c")
 class RDListingPage(BaseNPSPPage, ListingPage):
     object_name = "npe03__Recurring_Donation__c"
 
+    @capture_screenshot_on_error
+    def populate_rd2_modal_form(self, **kwargs):
+        """Populate the RD2 modal form fields with the respective fields and values"""
+        for key, value in kwargs.items():
+            if key in ("Amount", "Number of Planned Installments"):
+                locator = npsp_lex_locators["erd"]["modal_input_field"].format(key)
+                if self.npsp.check_if_element_exists(locator):
+                    self.selenium.set_focus_to_element(locator)
+                    self.salesforce._populate_field(locator, value)
+            if key in ("Account", "Contact"):
+                self.salesforce.populate_lookup_field(key, value)
+            else:
+                self.npsp.select_value_from_rd2_modal_dropdown(key,value)
 
 
 @pageobject("Details", "npe03__Recurring_Donation__c")
-class RDDetailPage(BaseNPSPPage,DetailPage ):
+class RDDetailPage(BaseNPSPPage, DetailPage):
     object_name = "npe03__Recurring_Donation__c"
-    
+
     def _is_current_page(self):
         """ Verify we are on the Account detail page
             by verifying that the url contains '/view'
         """
-        self.selenium.wait_until_location_contains("/view", timeout=60, message="Record view did not open in 1 min")
-        self.selenium.location_should_contain("/lightning/r/npe03__Recurring_Donation__c/",message="Current page is not a Recurring Donations record view")
-    
+        self.selenium.wait_until_location_contains(
+            "/view", timeout=60, message="Record view did not open in 1 min"
+        )
+        self.selenium.location_should_contain(
+            "/lightning/r/npe03__Recurring_Donation__c/",
+            message="Current page is not a Recurring Donations record view",
+        )
```

Comment:
based on recent observations, I suggest adding another check to wait for something to load on the page as we are seeing page load issues in some cases.

Reply:
Agree, Added an extra check